daphne/src/roles_test.rs

```diff
@@ -578,21 +579,30 @@ async fn e2e() {
         .unwrap();
     leader.put_out_shares(task_id, out_shares).await.unwrap();
 
-    // TODO(nakatsuka-y) Call http_post_collect to post a collect job.
-    //                   Assume for now that the leader has already picked a collect job.
+    // Collector: Create a CollectReq.
     let collect_req = CollectReq {
         task_id: task_id.clone(),
         batch_interval: task_config.current_batch_window(now),
         agg_param: Vec::default(),
     };
+    let req = DapRequest {
+        media_type: Some(MEDIA_TYPE_COLLECT_REQ),
+        payload: collect_req.get_encoded(),
+        url: task_config.helper_url.join("/collect").unwrap(),
+        sender_auth: Some(BearerToken::from(COLLECTOR_BEARER_TOKEN.to_string())),
+    };
+
+    // Leader: Handle the CollectReq received from Collector.
+    leader.http_post_collect(&req).await.unwrap();
```

Comment:
Test that the  collect URI encodes `collect_id`.

Reply:
I've implemented it in the new commits.